packages/block-library/src/paragraph/style.scss

```diff
@@ -1,46 +1,44 @@
-p {
-	&.is-small-text {
-		font-size: 14px;
-	}
+.is-small-text {
```

Comment:
Isn't this selector a bit _too_ generic? What about other blocks that use a class with this same name? Shouldn't this class be either namespaced (e.g. `.wp-block-paragraph__is-small-text` or changed to something along the lines of `.wp-block-paragraph.is-small-text`?

Reply:
Probably should've been `.wp-block-paragraph__is-small-text`. 
In the case of this PR though, does the `p` qualifier really make it any less generic? 

I think `.wp-block-paragraph.is-small-text` is too specific for a core style. Especially with something like `font-size` which is likely one of the first things that a theme will override.

Reply:
For the record though, I would prefer `.wp-block-paragraph.is-small-text` over `p.is-small-text`. Since there are more appropriate hooks in place for themes to override the Paragraph Block font-sizes.

As for being generic, the one that sticks out to me here is `.has-background`. That's not even "text" specific and we're adding some hefty padding here. Is that class used in any other blocks?

Reply:
Dang! This is going to have to be fixed. The button also uses `.has-background` and is getting the padding from the Paragraph block CSS. Guess We're going to have to increase specificity for the paragraph block rather than decrease. 

@kjellr How should I go about fixing? A new PR?

Reply:
Ah yes, good catch. I missed that selector. Open up a new PR and I'll take a look shortly.

Reply:
I think the `has-background` and `has-text-color` selectors are the only ones that definitely should be changed for now. I don't believe the text size and dropcap ones are used elsewhere.

Reply:
Had a few minutes, so I opened #13821 to take care of this. 👍

Reply:
Thanks @kjellr! 
Totally my fault. The PR was originally for the Dropcap and sort of morphed into being about specificity. I didn't put as much thought into it as I should have.

I agree that the other selectors don't look as problematic as `has-background`. Even if something like `is-small-text` is used in other blocks, we're only setting the font size, which would be expected for such a class name and would likely be the same or overridden by the other block.

It's the `padding` on a generic "background" selector that made this troublesome. 
It could be debated that the padding style is more appropriate in `theme.css`. But that's a topic for another day. :smile:

Reply:
You're all amazing. Thanks for the hard work here.

Reply:
> Totally my fault. The PR was originally for the Dropcap and sort of morphed into being about specificity. I didn't put as much thought into it as I should have.

No problem at all! It's all set now. Thanks again for your contribution, and thanks @ZebulanStanphill for starting the discussion that led to us catching this. 👍